frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+    if (ckEditorInstance) {
+      const data = ckEditorInstance.getData({ trim: false });
+      if (data.length > 0) {
+        localStorage.setItem(this.localStorageKey, data);
+      }
+    }
+  }
+
+  private populateRescuedEditorContent() {
+    const rescuedEditorContent = localStorage.getItem(this.localStorageKey);
+    if (rescuedEditorContent) {
+      this.openEditorWithInitialData(rescuedEditorContent);
+      localStorage.removeItem(this.localStorageKey);
+    }
+  }
+
+  private handleInitialScroll() {
+    if (window.location.hash.includes('#activity-')) {
+      this.scrollToActivity();
+    } else if (this.sortingValue === 'asc') {
+      this.scrollToBottom();
+    }
+  }
+
+  private scrollToActivity() {
+    const activityId = window.location.hash.replace('#activity-', '');
+    const activityElement = document.getElementById(`activity-${activityId}`);
+    activityElement?.scrollIntoView({ behavior: 'smooth' });
+  }
+
+  private scrollToBottom() {
+    const scrollableContainer = jQuery(this.element).scrollParent()[0];
+    if (scrollableContainer) {
+      setTimeout(() => {
+        scrollableContainer.scrollTop = scrollableContainer.scrollHeight;
+      }, 400);
+    }
+  }
+
+  setFilterToOnlyComments() { this.filterValue = 'only_comments'; }
+  setFilterToOnlyChanges() { this.filterValue = 'only_changes'; }
+  unsetFilter() { this.filterValue = ''; }
+
+  private getCkEditorInstance():ICKEditorInstance | null {
+    const AngularCkEditorElement = this.element.querySelector('opce-ckeditor-augmented-textarea');
+    return AngularCkEditorElement ? jQuery(AngularCkEditorElement).data('editor') as ICKEditorInstance : null;
+  }
+
+  private getInputContainer():HTMLElement | null {
+    return this.element.querySelector('#input-container');
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMobile():boolean {
+    return window.innerWidth < 1279;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isSmViewPort():boolean {
+    return window.innerWidth < 543;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMdViewPort():boolean {
+    return window.innerWidth >= 543 && window.innerWidth < 1279;
+  }
+
+  private addEventListenersToCkEditorInstance() {
+    const editor = this.getCkEditorInstance();
+    if (editor) {
+      this.addKeydownListener(editor);
+      this.addKeyupListener(editor);
+      this.addBlurListener(editor);
+    }
+  }
+
+  private addKeydownListener(editor:ICKEditorInstance) {
+    editor.listenTo(
+      editor.editing.view.document,
+      'keydown',
+      (event, data) => {
+        // taken from op-ck-editor.component.ts
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+        if ((data.ctrlKey || data.metaKey) && data.keyCode === KeyCodes.ENTER) {
+          void this.onSubmit();
```

Comment:
Why did you need to duplicate this code here? can't we use an output from the op-ckeditor component?

Reply:
Done